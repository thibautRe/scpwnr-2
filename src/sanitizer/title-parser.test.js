const parser = require('./title-parser')

const fixtures = [
  'Black Coast - TRNDSTTR (Lucian Remix) [feat. M. Maggie]',
  'Maff Boothroyd - Miss You 🍉',
  'Blank (feat. Kennedi)',
  'BIJOU - Memories',
  'Tails X Inverness – Skeleton Ft. Nevve (Voorhees Remix)',
]

test('title parser', () => {
  fixtures.forEach((title) => expect(parser(title)).toMatchSnapshot(title))
})
