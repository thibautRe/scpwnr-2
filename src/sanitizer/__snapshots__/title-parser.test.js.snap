// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title parser: BIJOU - Memories 1`] = `
Object {
  "artist": "BIJOU",
  "name": "Memories",
}
`;

exports[`title parser: Black Coast - TRNDSTTR (Lucian Remix) [feat. M. Maggie] 1`] = `
Object {
  "artist": "Black Coast",
  "ft": "M. Maggie",
  "name": "TRNDSTTR",
  "remixType": "remix",
  "remixedBy": "Lucian",
}
`;

exports[`title parser: Blank (feat. Kennedi) 1`] = `
Object {
  "ft": "Kennedi",
  "name": "Blank",
}
`;

exports[`title parser: Maff Boothroyd - Miss You üçâ 1`] = `
Object {
  "artist": "Maff Boothroyd",
  "name": "Miss You üçâ",
}
`;

exports[`title parser: Tails X Inverness ‚Äì Skeleton Ft. Nevve (Voorhees Remix) 1`] = `
Object {
  "artist": "Tails X Inverness",
  "name": "Skeleton Ft. Nevve",
  "remixType": "remix",
  "remixedBy": "Voorhees",
}
`;
